### PR TITLE
libtommath: added shared library, replaced deprecated test_standalone with test

### PR DESCRIPTION
**Pull request by andrew-aladev**
- [*] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [*] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [*] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libtommath.rb [hunk 1]
```diff
@@ -28,10 +30,10 @@ class Libtommath < Formula
   def install
     ENV["DESTDIR"] = prefix
 
-    system "make"
-    system "make", "test_standalone"
+    system "make", "-f", "makefile.shared"
+    system "make", "test"
     include.install Dir["tommath*.h"]
-    lib.install "libtommath.a"
+    lib.install Dir[".libs/libtommath*{.a,.dylib}"]
```
Comment (carlocab) [inline Formula/libtommath.rb:36]:
```suggestion
    lib.install Dir[".libs/libtommath*.a"]
    lib.install Dir["./libs/#{shared_library("libtommath", "*")}"]
```
I think this should make it portable for Linuxbrew. Oh, and can you increment the `revision` to `2`? Squash your commits together (if you have multiple) when you're done. Thanks!

File: Formula/libtommath.rb [hunk 2]
```diff
@@ -15,6 +15,8 @@ class Libtommath < Formula
     sha256 cellar: :any_skip_relocation, high_sierra:   "26e39af069485ef58c3517fb765db3a5e8dba0f253aac3d0d5968ff2a35e595b"
   end
 
+  depends_on "libtool" => :build
```
Comment (carlocab) [inline Formula/libtommath.rb:18]: Actually, this just occurred to me: Homebrew `libtool` binaries are named `glibtool` and `glibtoolize`. Does `make` not need help finding them? 

I'm actually wondering if it's possible that `make` didn't find them, but was able to build everything just fine with system `libtool`.

Comment (andrew-aladev) [reply Formula/libtommath.rb:18]: `libtommath` makefile includes [special case for darwin platform](https://github.com/libtom/libtommath/blob/develop/makefile.shared).

```
ifeq ($(PLATFORM), Darwin)
  LIBTOOL:=glibtool
else
  LIBTOOL:=libtool
endif
```

We can see that `libtool` is required.

**Comment by andrew-aladev** [comment]
It looks like new formula works good, please review, thank you.

File: Formula/libtommath.rb [hunk 3]
```diff
@@ -28,10 +30,10 @@ class Libtommath < Formula
   def install
     ENV["DESTDIR"] = prefix
 
-    system "make"
-    system "make", "test_standalone"
+    system "make", "-f", "makefile.shared"
+    system "make", "test"
     include.install Dir["tommath*.h"]
-    lib.install "libtommath.a"
+    lib.install Dir[".libs/libtommath*.a", ".libs/#{shared_library("libtommath", "*")}"]
```
Comment (carlocab) [inline Formula/libtommath.rb:36]: Actually, now that I've had a look at `makefile` and `makefile.shared`, is there any reason we don't do `make install` here?

**Review by carlocab — APPROVED** [review]
Thanks, @andrew-aladev.

**Pull request closed without merging**